mock/soap/src/main/java/io/gatehill/imposter/plugin/soap/parser/WsdlRelativeXsdEntityResolver.kt

```diff
@@ -0,0 +1,31 @@
+package io.gatehill.imposter.plugin.soap.parser
+
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.xml.sax.EntityResolver
+import org.xml.sax.InputSource
+import java.io.File
+import java.io.FileInputStream
+import java.nio.file.Files
+
+/**
+ * EntityResolver to resolve XSDs included relative to the WSDL file.
+ */
+class WsdlRelativeXsdEntityResolver(
+    private val wsdlDir: File
+) : EntityResolver {
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    override fun resolveEntity(publicId: String?, systemId: String?): InputSource? {
+        logger.trace("Resolve {} relative to path {}", systemId, wsdlDir)
+        systemId ?: return null
+
+        // in certain occasions, XMLBeans prefixes the systemId (see StscState) -> strip it off
+        val xsdFile = File(wsdlDir.absoluteFile, systemId.replace("project://local/", ""))
+        if (Files.isRegularFile(xsdFile.toPath()) && xsdFile.name.lowercase().endsWith(".xsd")) {
+            logger.debug("Resolved XSD {} relative to WSDL path", xsdFile)
+            return InputSource(FileInputStream(xsdFile))
```

Comment:
Do we know if `InputSource` closes the stream it's provided, or is there a resource leak here?

Reply:
I have not checked the places where it's getting called. But when looking at other examples (e.g. com.sun.rowset.internal.XmlResolver, org.apache.log4j.xml.Log4jEntityResolver, ...) none of them closes the stream.

Reply:
So, from the `InputSource` JavaDoc:

```java
/*
 * ...
 * standard processing of both byte and
 * character streams is to close them on as part of end-of-parse cleanup,
 * so applications should not attempt to re-use such streams after they
 * have been handed to a parser.
 * ...
```

My read of this is that we don't need to manage the lifecycle of the stream as it's done by the parser using the resolver.

Reply:
I also verified when running the tests. I can confirm the `InputStream` is getting closed.

Reply:
Brilliant, thank you.